blint/sbom.py

```diff
@@ -598,18 +598,28 @@ def process_rust_dependencies(rust_deps: list) -> list[Component]:
         list: New component list
     """
     components = []
+    idx_to_purl = {}
+    for idx, dep in enumerate(rust_deps):
+        idx_to_purl[idx] = f"""pkg:cargo/{dep["name"]}@{dep["version"]}"""
     for dependency in rust_deps:
         purl = f"""pkg:cargo/{dependency["name"]}@{dependency["version"]}"""
+        purl_qualifer = f"""?repository={dependency.get("source")}""" if dependency.get("source", "") != "crates.io" else ""
         comp = Component(
             type=Type.library,
             name=dependency["name"],
             version=dependency["version"],
-            purl=purl,
+            purl=f"{purl}{purl_qualifer}",
             scope=Scope.required,
-            evidence=create_component_evidence(dependency["name"], 1.0)
+            evidence=create_component_evidence(dependency["name"], 0.8)
```

Comment:
Removed confidence slightly, since cargo-auditable isn't perfect and could over-estimate dependencies.